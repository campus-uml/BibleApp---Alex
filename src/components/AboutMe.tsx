import {
  Card,
  CardContent,
  CardHeader,

} from "@/components/ui/card";
import { Avatar, AvatarFallback, AvatarImage } from "@/components/ui/avatar";
import { Badge } from "@/components/ui/badge";
import { FacebookIcon, GithubIcon, LinkedinIcon } from "lucide-react";

export const AboutMe = () => {
  return (
    <Card className="w-full max-w-md mx-auto ">
      <CardHeader className="text-center">
      </CardHeader>
      <CardContent className="flex flex-col items-center space-y-4">
        <Avatar className="w-24 h-24">
          <AvatarImage src="/avatar.jpg" alt="Foto del desarrollador" />
          <AvatarFallback>
            <img
              src="https://i.pinimg.com/736x/3e/f9/5e/3ef95eb2be325c976ac5e33d3efb1224.jpg"
              alt="Perfil"
            />
          </AvatarFallback>
        </Avatar>
        <div className="text-center">
          <h2 className="text-xl font-semibold">Eddy Alexis Talavera</h2>
          <p className="text-sm text-muted-foreground">Desarrollador junior</p>
        </div>
        <div className="flex space-x-2">
          <Badge variant="secondary">React</Badge>
          <Badge variant="secondary">TypeScript</Badge>
          <Badge variant="secondary">Tailwind CSS</Badge>
        </div>
        <p className="text-sm text-center max-w-xs">
          Apasionado por crear aplicaciones web intuitivas y eficientes.
          Especializado en React y TypeScript.
        </p>
        <div className="flex space-x-4">
          <a
            href="https://github.com/Alex200207"
            target="_blank"
            rel="noopener noreferrer"
            className="text-muted-foreground hover:text-primary"
          >
            <GithubIcon className="w-6 h-6" />
          </a>
          <a
            href="https://www.linkedin.com/in/alex-talavera-57a8b12a2/"
            target="_blank"
            rel="noopener noreferrer"
            className="text-muted-foreground hover:text-primary"
          >
            <LinkedinIcon className="w-6 h-6" />
          </a>
          <a
            href="https://www.facebook.com/alexis.talavera.503?locale=es_LA"
            target="_blank"
            rel="noopener noreferrer"
            className="text-muted-foreground hover:text-primary"
          >
            <FacebookIcon className="w-6 h-6" />
          </a>
        </div>
      </CardContent>
    </Card>
  );
};
