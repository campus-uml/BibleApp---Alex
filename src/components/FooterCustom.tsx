import { Github } from "lucide-react";

const FooterCustom = () => {
  return (
    <footer className="bg-background text-foreground border-t">
      <div className="container mx-auto px-4 py-6">
        <div className="flex flex-col md:flex-row justify-around items-center">
          <div className="flex flex-col items-center md:items-start mb-4 md:mb-0">
            <p className="text-sm text-muted-foreground">
              &copy; {new Date().getFullYear()} BibleApp
            </p>
            <a
              href="https://github.com/Alex200207"
              target="_blank"
              rel="noopener noreferrer"
              className="flex items-center gap-2 mt-2 text-foreground hover:text-primary transition-colors"
            >
              <Github className="w-5 h-5" />
              <span className="font-medium">Alex Talavera</span>
            </a>
          </div>
          <div className="flex items-center gap-4">
            <a
              href="/privacy-policy"
              className="text-sm text-muted-foreground hover:text-foreground transition-colors"
            >
              Privacy Policy
            </a>
            <a
              href="/terms-of-service"
              className="text-sm text-muted-foreground hover:text-foreground transition-colors"
            >
              Terms of Service
            </a>
          </div>
        </div>
      </div>
    </footer>
  );
};

export default FooterCustom;
