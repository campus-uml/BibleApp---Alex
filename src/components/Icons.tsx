import { BookIcon as Bible, Loader2 } from "lucide-react"

export const Icons = {
  bible: Bible,
  spinner: Loader2,
}

