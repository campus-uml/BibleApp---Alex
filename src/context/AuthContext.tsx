import { createContext, useContext, useEffect, useState } from "react";
import { supabase } from "..//constants/api";
import { User } from "@/types";

interface AuthContextType {
  user: User | null;
  logout: () => void;
}

const AuthContext = createContext<AuthContextType | null>(null);

export const AuthProvider = ({ children }: { children: React.ReactNode }) => {
  const [user, setUser] = useState<User | null>(null);

  useEffect(() => {
    const fetchUser = async () => {
      const { data } = await supabase.auth.getUser();
      if (data.user) setUser(data.user);
    };

    fetchUser();

    const { data: listener } = supabase.auth.onAuthStateChange((_event, session) => {
      setUser(session?.user || null);
    });

    return () => {
      listener?.subscription.unsubscribe();
    };
  }, []);

  const logout = async () => {
    await supabase.auth.signOut();
    setUser(null);
  };

  return (
    <AuthContext.Provider value={{ user, logout }}>
      {children}
    </AuthContext.Provider>
  );
};

// eslint-disable-next-line react-refresh/only-export-components
export const useAuth = () => {
  return useContext(AuthContext);
};
