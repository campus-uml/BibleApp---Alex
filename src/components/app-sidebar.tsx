import * as React from "react";
import { GalleryVerticalEnd, Search } from "lucide-react";
import {
  Sidebar,
  SidebarContent,
  SidebarGroup,
  SidebarHeader,
  SidebarMenu,
  SidebarMenuButton,
} from "@/components/ui/sidebar";
import { TabBible } from "./TabBible";
import { SearchInput } from "../components/searchInput";
import { Button } from "@/components/ui/button";
import {
  Tooltip,
  TooltipContent,
  TooltipProvider,
  TooltipTrigger,
} from "@/components/ui/tooltip";

export function AppSidebar({ ...props }: React.ComponentProps<typeof Sidebar>) {
  const [showSearch, setShowSearch] = React.useState(false);

  return (
    <Sidebar {...props}>
      <SidebarHeader className="flex flex-col space-y-2 p-2">
        <div className="flex items-center justify-between">
          <SidebarMenuButton size="sm" asChild>
            <a href="#" className="flex items-center space-x-2">
              <div className="flex aspect-square size-8 items-center justify-center rounded-lg bg-sidebar-primary text-sidebar-primary-foreground">
                <GalleryVerticalEnd className="size-4" />
              </div>
              <span className="font-semibold">BibleApp</span>
            </a>
          </SidebarMenuButton>
          <TooltipProvider>
            <Tooltip>
              <TooltipTrigger asChild>
                <Button
                  variant="ghost"
                  size="icon"
                  onClick={() => setShowSearch(!showSearch)}
                >
                  <Search className="h-4 w-4" />
                </Button>
              </TooltipTrigger>
              <TooltipContent>
                <p>Buscar en la Biblia</p>
              </TooltipContent>
            </Tooltip>
          </TooltipProvider>
        </div>
        {showSearch && <SearchInput />}
      </SidebarHeader>
      <SidebarContent>
        <SidebarGroup>
          <SidebarMenu>
            <TabBible />
          </SidebarMenu>
        </SidebarGroup>
      </SidebarContent>
    </Sidebar>
  );
}
