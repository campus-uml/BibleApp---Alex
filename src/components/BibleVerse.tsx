import { Tabs, TabsContent, TabsList, TabsTrigger } from "@/components/ui/tabs";
import { ScrollArea, ScrollBar } from "@/components/ui/scroll-area";
import { useBible } from "../context/BIbleContext";
import { Copy, Heart, ArrowLeft, Share, Play, Book } from "lucide-react";
import type { SearchResults, Verse } from "@/types";
import { Button } from "@/components/ui/button";
import { useActions } from "@/Hooks/useActions";

interface BibleVerseProps {
  searchResults: SearchResults | null;
  onClearSearch: () => void;
  addFavorite: (verse: Verse) => void;
  removeFavorite: (verseId: string) => void;
  favorites: Verse[];
}

export const BibleVerse = ({
  searchResults,
  onClearSearch,
  addFavorite,
  favorites,
  removeFavorite,
}: BibleVerseProps) => {
  const {
    bibleVerseChapters,
    chapterVerses,
    loadChapterVerses,
    scrollAreaRef,
    selectedBook,
    bibleVerse,
  } = useBible();
  const { copyToClipboard, shareToWhatsApp } = useActions();

  const bookName = (selectedBook: string) => {
    const book = bibleVerse.find((book) => book.id === selectedBook);
    return book ? book.name : "";
  };

  return (
    <div className="min-h-screen bg-gray-50">
      <Tabs
        defaultValue="1"
        className="w-full max-w-[21.2rem] md:max-w-6xl mx-auto"
      >
        <div className="sticky top-0 bg-gray-50 z-10 pb-4">
          {searchResults ? (
            <div className="flex items-center justify-between mb-4">
              <Button
                variant="outline"
                onClick={onClearSearch}
                className="flex items-center gap-2"
              >
                <ArrowLeft size={16} />
                Volver
              </Button>
              <h2 className="text-xl font-semibold text-gray-800">
                Resultados de búsqueda
              </h2>
            </div>
          ) : (
            <>
              <h1 className="text-2xl font-bold text-center text-gray-800 mb-4">
                {bookName(selectedBook)}
              </h1>
              {bibleVerseChapters.length > 0 ? (
                <TabsList className="h-12 items-center justify-start rounded-md bg-muted p-1 text-muted-foreground w-full overflow-hidden">
                  <ScrollArea
                    ref={scrollAreaRef}
                    className="w-full overflow-x-auto whitespace-nowrap scrollbar-thin scrollbar-thumb-gray-300 scrollbar-track-gray-100"
                  >
                    <div className="flex items-center gap-2 py-1">
                      {bibleVerseChapters.map((chapter) => (
                        <TabsTrigger
                          key={chapter.id}
                          value={chapter.number.toString()}
                          onClick={() => loadChapterVerses(chapter.id)}
                          className="ring-offset-background focus-visible:ring-ring data-[state=active]:bg-background data-[state=active]:text-foreground px-3 py-1.5 rounded-md"
                        >
                          {chapter.number}
                        </TabsTrigger>
                      ))}
                    </div>
                    <ScrollBar orientation="horizontal" />
                  </ScrollArea>
                </TabsList>
              ) : (
                <div className="text-center py-8">
                  <Book className="w-12 h-12 text-slate-300 mx-auto mb-3" />
                  <p className="text-lg font-semibold text-slate-600">
                    No hay versículos aún
                  </p>
                  <p className="text-slate-500 mb-4">
                    Selecciona en la parte superior para comenzar
                  </p>
                </div>
              )}
            </>
          )}
        </div>

        {searchResults &&
        searchResults.verses &&
        searchResults.verses.length > 0 ? (
          <div className="space-y-4 mt-4">
            {searchResults.verses.map((verse, index) => (
              <div key={index} className="bg-white shadow-md rounded-lg p-2">
                <h3 className="text-sm text-slate-500 font-semibold mb-2">
                  {verse.id}
                </h3>
                <p className="text-base text-gray-700">{verse.text}</p>
                <div className="flex justify-end gap-4 mt-4">
                  <button
                    onClick={() =>
                      favorites.some((fav) => fav.id === verse.id)
                        ? removeFavorite(verse.id)
                        : addFavorite({ ...verse, reference: verse.id })
                    }
                    className="mt-2 focus:outline-none"
                    aria-label={
                      favorites.some((fav) => fav.id === verse.id)
                        ? "remover"
                        : "agregar"
                    }
                  >
                    <Heart
                      className={`w-6 h-6 ${
                        favorites.some((fav) => fav.id === verse.id)
                          ? "text-red-500"
                          : "text-gray-900"
                      }`}
                    />
                  </button>
                  <button className="text-gray-600 hover:text-blue-500 transition-colors">
                    <Play size={20} />
                  </button>
                  <button
                    onClick={() => copyToClipboard(verse.text)}
                    className="text-gray-600 hover:text-green-500 transition-colors"
                  >
                    <Copy size={20} />
                  </button>
                  <button
                    onClick={() => shareToWhatsApp(verse.text)}
                    className="text-gray-600 hover:text-purple-500 transition-colors"
                  >
                    <Share size={20} />
                  </button>
                </div>
              </div>
            ))}
          </div>
        ) : (
          <div className="space-y-4 mt-4">
            {bibleVerseChapters.map((chapter) => (
              <TabsContent
                key={chapter.id}
                value={chapter.number.toString()}
                className="bg-white rounded-lg shadow-sm p-2"
              >
                <h2 className="text-xl text-center font-semibold mb-4 text-gray-800">
                  Capítulo {chapter.number}
                </h2>
                <div className="space-y-4">
                  {chapterVerses.map((verse, index) => (
                    <div
                      key={index}
                      className="bg-white shadow-md rounded-lg p-4"
                    >
                      <h3 className="text-sm text-slate-500 font-semibold mb-2">
                        {verse.reference}
                      </h3>
                      <p className="text-gray-700 leading-relaxed italic">
                        {verse.text}
                      </p>
                      <div className="flex justify-end gap-4 mt-4">
                        <button
                          onClick={() =>
                            favorites.some((fav) => fav.id === verse.id)
                              ? removeFavorite(verse.id)
                              : addFavorite({ ...verse, reference: verse.id })
                          }
                          className="mt-2 focus:outline-none"
                          aria-label={
                            favorites.some((fav) => fav.id === verse.id)
                              ? "remover"
                              : "agregar"
                          }
                        >
                          <Heart
                            size={20}
                            className={` ${
                              favorites.some((fav) => fav.id === verse.id)
                                ? "text-red-500"
                                : "text-gray-900"
                            }`}
                          />
                        </button>
                        <button className="text-gray-600 hover:text-blue-500 transition-colors">
                          <Play size={20} />
                        </button>
                        <button
                          onClick={() => copyToClipboard(verse.text)}
                          className="text-gray-600 hover:text-green-500 transition-colors"
                        >
                          <Copy size={20} />
                        </button>
                        <button
                          onClick={() => shareToWhatsApp(verse.text)}
                          className="text-gray-600 hover:text-purple-500 transition-colors"
                        >
                          <Share size={20} />
                        </button>
                      </div>
                    </div>
                  ))}
                </div>
              </TabsContent>
            ))}
          </div>
        )}
      </Tabs>
    </div>
  );
};

export default BibleVerse;
