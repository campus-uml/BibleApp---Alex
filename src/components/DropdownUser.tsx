import { Button } from "@/components/ui/button";
import {
  DropdownMenu,
  DropdownMenuContent,
  DropdownMenuItem,
  DropdownMenuLabel,
  DropdownMenuSeparator,
  DropdownMenuTrigger,
} from "@/components/ui/dropdown-menu";
import { Avatar, AvatarFallback, AvatarImage } from "@/components/ui/avatar";
import LogoutButton from "@/components/login/LogoutButton";
import { Settings, UserIcon } from "lucide-react";
import { useAuth } from "@/context/AuthContext";

interface DropdownUserProps {
  avatarUrl: string | null;
}

export function DropdownUser({ avatarUrl }: DropdownUserProps) {
  const { user } = useAuth();

  const displayName = user?.user_metadata.full_name || user?.user_metadata.name;

  console.log("user", user);
  return (
    <DropdownMenu>
      <DropdownMenuTrigger asChild>
        <Button
          variant="ghost"
          size="icon"
          className="rounded-full hover:bg-accent"
        >
          <Avatar className="h-8 w-8 transition-transform hover:scale-110">
            <AvatarImage src={avatarUrl ?? undefined} alt="User Avatar" />
            <AvatarFallback className="bg-primary text-primary-foreground">
            {user?.email?.[0].toUpperCase() ?? "?"}
            </AvatarFallback>
          </Avatar>
        </Button>
      </DropdownMenuTrigger>
      <DropdownMenuContent align="end" className="w-56">
        <div className="flex items-center justify-start gap-2 p-2">
          <Avatar className="h-8 w-8">
            <AvatarImage src={avatarUrl ?? undefined} alt="User Avatar" />
            <AvatarFallback className="bg-primary text-primary-foreground">
            {user?.email?.[0].toUpperCase() ?? "?"}
            </AvatarFallback>
          </Avatar>
          <div className="flex flex-col space-y-1">
            <DropdownMenuLabel className="font-medium">
              {displayName}
            </DropdownMenuLabel>
            <p className="text-xs text-muted-foreground">
              @{user?.user_metadata.user_name || user?.email}
            </p>
          </div>
        </div>
        <DropdownMenuSeparator />
        <DropdownMenuItem>
          <UserIcon className="mr-2 h-4 w-4" />
          <span>Perfil</span>
        </DropdownMenuItem>
        <DropdownMenuItem>
          <Settings className="mr-2 h-4 w-4" />
          <span>Configuraci√≥n</span>
        </DropdownMenuItem>
        <DropdownMenuSeparator />
        <DropdownMenuItem asChild>
          <LogoutButton />
        </DropdownMenuItem>
      </DropdownMenuContent>
    </DropdownMenu>
  );
}
