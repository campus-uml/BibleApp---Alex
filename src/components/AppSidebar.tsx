import * as React from "react";
import { GalleryVerticalEnd, Search } from "lucide-react";
import {
  Sidebar,
  SidebarContent,
  SidebarGroup,
  SidebarHeader,
  SidebarMenu,
  SidebarMenuButton,
} from "@/components/ui/sidebar";
import { TabBible } from "./TabBible";
import { Button } from "@/components/ui/button";
import {
  Tooltip,
  TooltipProvider,
  TooltipTrigger,
} from "@/components/ui/tooltip";
import { DropdownUser } from "./DropdownUser";
import { useState } from "react";
import { useAuth } from "@/context/AuthContext";
import SearchBook from "./SearchBook";

export function AppSidebar({ ...props }: React.ComponentProps<typeof Sidebar>) {
  const [showSearch, setShowSearch] = useState(false);
  const [searchTerm, setSearchTerm] = useState("");
  const { avatarUrl, user } = useAuth();

  const handleSearch = (term: string) => {
    setSearchTerm(term);
  };

  return (
    <Sidebar {...props}>
      <SidebarHeader className="flex flex-col space-y-2 p-2">
        <div className="flex items-center justify-between">
          <SidebarMenuButton size="sm" asChild>
            <a href="#" className="flex items-center space-x-2">
              <div className="flex aspect-square size-8 items-center justify-center rounded-lg bg-sidebar-primary text-sidebar-primary-foreground">
                <GalleryVerticalEnd className="size-4" />
              </div>
              <span className="font-semibold">BibleApp</span>
            </a>
          </SidebarMenuButton>
          <div className="flex items-center space-x-2">
            <TooltipProvider>
              <Tooltip>
                <TooltipTrigger asChild>
                  <Button
                    variant="ghost"
                    size="icon"
                    onClick={() => setShowSearch(!showSearch)}
                  >
                    <Search className="h-4 w-4" />
                  </Button>
                </TooltipTrigger>
              </Tooltip>
            </TooltipProvider>
            <DropdownUser avatarUrl={avatarUrl} user={user} />
          </div>
        </div>
        {showSearch && <SearchBook onSearch={handleSearch} />}
      </SidebarHeader>
      <SidebarContent>
        <SidebarGroup>
          <SidebarMenu>
            <TabBible searchTerm={searchTerm} />
          </SidebarMenu>
        </SidebarGroup>
      </SidebarContent>
    </Sidebar>
  );
}
