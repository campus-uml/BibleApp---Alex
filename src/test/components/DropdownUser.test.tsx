import { describe, it, expect, vi } from "vitest";
import { render, screen } from "@testing-library/react";
import { DropdownUser } from "../../components/DropdownUser";
import { AuthProvider } from "@/context/AuthContext";

// Mock the useAuth hook
vi.mock("@/context/AuthContext", () => ({
  useAuth: () => ({
    user: {
      user_metadata: {
        full_name: "John Doe",
        user_name: "johndoe",
      },
      email: "john@example.com",
    },
  }),
  AuthProvider: ({ children }: { children: React.ReactNode }) => (
    <div>{children}</div>
  ),
}));

vi.mock("@/components/ui/mode-toggle", () => ({
  ModeToggle: () => <div data-testid="mode-toggle">Mode Toggle</div>,
}));

vi.mock("@/components/home/AboutModal", () => ({
  AboutModal: () => <div data-testid="about-modal">About Modal</div>,
}));

vi.mock("@/components/login/LogoutButton", () => ({
  __esModule: true,
  default: () => <button data-testid="logout-button">Logout</button>,
}));

describe("DropdownUser", () => {
  it("renders fallback avatar when no avatarUrl is provided", () => {
    const { asFragment } = render(
      <AuthProvider>
        <DropdownUser avatarUrl={null} />
      </AuthProvider>
    );
    const avatarButton = screen.getByRole("button");
    expect(avatarButton).toBeInTheDocument();
    expect(avatarButton).toHaveTextContent("J");
    expect(asFragment()).toMatchSnapshot();
  });
});
