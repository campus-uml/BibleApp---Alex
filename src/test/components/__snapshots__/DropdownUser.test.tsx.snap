// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`DropdownUser > renders fallback avatar when no avatarUrl is provided 1`] = `
<DocumentFragment>
  <div>
    <button
      aria-expanded="false"
      aria-haspopup="menu"
      class="inline-flex items-center justify-center gap-2 whitespace-nowrap text-sm font-medium focus-visible:outline-none focus-visible:ring-1 focus-visible:ring-ring disabled:pointer-events-none disabled:opacity-50 [&_svg]:pointer-events-none [&_svg]:size-4 [&_svg]:shrink-0 hover:text-accent-foreground h-9 w-9 rounded-full hover:bg-accent transition-all duration-300 ease-in-out"
      data-state="closed"
      id="radix-:r0:"
      type="button"
    >
      <span
        class="relative flex shrink-0 overflow-hidden rounded-full h-8 w-8 transition-transform duration-300 ease-in-out hover:scale-110"
      >
        <span
          class="flex h-full w-full items-center justify-center rounded-full bg-primary text-primary-foreground"
        >
          J
        </span>
      </span>
    </button>
  </div>
</DocumentFragment>
`;
